rules/msk_topic_config_comments.go

```diff
@@ -271,10 +329,73 @@ func (r *MSKTopicConfigCommentsRule) buildDurationComment(
 		return "", nil
 	}
 
-	baseComment := configValueInfo.baseComment
+	return buildCommentForMillis(timeMillis, configValueInfo.baseComment), nil
+}
+
+func (r *MSKTopicConfigCommentsRule) buildDataSizeComment(
+	runner tflint.Runner,
+	dataPair hcl.KeyValuePair,
+	configValueInfo configValueCommentInfo,
+) (string, error) {
+	var dataVal string
+	diags := gohcl.DecodeExpression(dataPair.Value, nil, &dataVal)
+	if diags.HasErrors() {
+		return "", diags
+	}
+
+	if dataVal == configValueInfo.infiniteValue {
+		return fmt.Sprintf("# %s unlimited data", configValueInfo.baseComment), nil
+	}
+
+	byteVal, err := strconv.Atoi(dataVal)
+	if err != nil {
+		if configValueInfo.issueWhenInvalid {
+			issueMsg := fmt.Sprintf(
+				"%s must have a valid integer value expressed in bytes",
+				configValueInfo.key,
+			)
+			err := runner.EmitIssue(r, issueMsg, dataPair.Value.Range())
+			if err != nil {
+				return "", fmt.Errorf("emitting issue: invalid data value: %w", err)
+			}
+		}
+
+		return "", nil
+	}
+
+	return buildCommentForBytes(byteVal, configValueInfo.baseComment), nil
+}
+
+func buildCommentForBytes(bytes int, baseComment string) string {
+	byteUnits, unit := determineByteUnits(bytes)
+
+	byteUnitsStr := strconv.FormatFloat(byteUnits, 'f', -1, 64)
+	return fmt.Sprintf("# %s %s%s", baseComment, byteUnitsStr, unit)
+}
+
+const (
+	bytesInOneKB = 1024
+	bytesInOneMB = 1024 * bytesInOneKB
+	bytesInOneGB = 1024 * bytesInOneMB
+)
+
+func determineByteUnits(bytes int) (float64, string) {
+	floatBytes := float64(bytes)
+	gbs := round(floatBytes / bytesInOneGB)
+	if gbs >= 1 {
+		return gbs, "GB"
+	}
+
+	mbs := round(floatBytes / bytesInOneMB)
+	if mbs >= 1 {
+		return mbs, "MB"
```

Comment:
can we use [binary prefixes](https://en.wikipedia.org/wiki/Binary_prefix) to avoid any ambiguity about the value

```suggestion
	bytesInOneKiB = 1024
	bytesInOneMiB = 1024 * bytesInOneKiB
	bytesInOneGiB = 1024 * bytesInOneMiB
)

func determineByteUnits(bytes int) (float64, string) {
	floatBytes := float64(bytes)
	gbs := round(floatBytes / bytesInOneGiB)
	if gbs >= 1 {
		return gbs, "GiB"
	}

	mbs := round(floatBytes / bytesInOneMiB)
	if mbs >= 1 {
		return mbs, "MiB"
```

Reply:
Yep, good idea

Reply:
Updated in https://github.com/utilitywarehouse/tflint-ruleset-kafka-config/pull/32/commits/04bb3a12841dc624423ad62097a2aab32dc7167e